nixos/modules/services/x11/redshift.nix

```diff
@@ -92,12 +119,20 @@ in {
     };
 
     extraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [];
-      example = [ "-v" "-m randr" ];
+      type = types.str;
+      default = "";
+      example = ''
+        ; Will be applied in "[redshift]" section
+        elevation-high=3
+        elevation-low=-6
+
+        ; Using another section
+        [randr]
+        screen=0
+      '';
```

Comment:
I am not sure if this is the best way to do it. I think that maybe accepting a map, something like:

```nix
{
  randr = {
    screen = "0";
  };
}
```

Maybe would be better but probably way more complicated (I don't have a clue on how to implement this in Nix). Also I don't know what to do in the conflict cases (`{ redshift = { ... }` since there is already a `[redshift]` section).

Reply:
BTW, if this way I did is ok I can document this breaking change in release notes.

Reply:
Look at the `toINI` generator. It writes dictionaries to various formats INI. I think this simplifies everything, because then you can leave everything as Nix sets until the very end where you write to the file.

Reply:
@sumnerevans Does it solve the conflict cases too :thinking: ? Like:

```nix
{
  foo =  {
    bar = "1";
    qux = "1";
  };
  foo = {
    baz = "2";
    qux = "3";
  };
}
```

```
[foo]
bar = "1"
baz = "2"
qux = "3"
```

?

Reply:
I think that the Nix `//` operator should do that... Like:
```
{
  foo =  {
    bar = "1";
    qux = "1";
  };
} // {
  foo = {
    baz = "2";
    qux = "3";
  };
}
```

then you can just write that using the `toINI` function. Not at my Nix computer (on my work computer right now), but I think that something like that should work.

Reply:
Answering my question:

```
nix-repl> lib.generators.toINI {} { foo = { bar = "qux"; }; foo = { baz = "quux"; }; }  
"[foo]\nbar=qux\nbaz=quux\n"
nix-repl> lib.generators.toINI {} { foo = { bar = "qux"; }; foo = { bar = "quux"; baz = "quux"; }; }    
error: attribute 'bar' at (string):1:35 already defined at (string):1:59
```

So defining the same root key is fine, redefining a key is not. I think this is ok I guess, will work sufficiently fine for my purposes.

Reply:
OK, that looks great! BTW, thanks for doing this, definitely better to have it as a config file IMO.